### PR TITLE
Remove compaction file longer than certain time

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
Clean up compaction files created before a configurable time.

### CODE REVIEW

File: ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java [hunk 1]
```diff
@@ -128,6 +123,47 @@ static void processCompactionLogs(String dir, String storeId, StoreKeyFactory st
...
+    }
+    // The list is in descending order, the most recent compaction logs are in the front, we have to reverse the list
+    for (int i = sortedFiles.size() - 1; i >= 0; i--) {
+      File compactionFile = sortedFiles.get(i);
+      long startTime = getStartTimeFromFile(compactionFile);
+      if (startTime >= beforeTime) {
+        break;
+      } else {
+        logger.info("{}: Removing compaction file: {}, cutoff time: {}", storeId, compactionFile.getName(), beforeTime);
+        compactionFile.delete();
```
Comment (JingQianCloud) [inline ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java:146]: Is there any change we are still writing the compaction file? Is there any way to check for that?

File: ambry-store/src/main/java/com/github/ambry/store/BlobStore.java [hunk 2]
```diff
@@ -1204,10 +1202,9 @@ void updateSealedStatus() {
               replicaId.getPartitionId(), index.getLogUsedCapacity(), sealThresholdBytesLow, sealThresholdBytesHigh);
         }
       }
-    } else if (resolvedReplicaSealStatus == ReplicaSealStatus.PARTIALLY_SEALED && (
-        (replicaStatusDelegates.size() > 1
-            && replicaSealStatus.getAndSet(ReplicaSealStatus.PARTIALLY_SEALED) != ReplicaSealStatus.PARTIALLY_SEALED)
-            || !replicaId.isPartiallySealed())) {
+    } else if (resolvedReplicaSealStatus == ReplicaSealStatus.PARTIALLY_SEALED && ((replicaStatusDelegates.size() > 1
```
[the remark this answers is not in the document]
Comment (justinlin-linkedin) [reply ambry-store/src/main/java/com/github/ambry/store/BlobStore.java:1205]: you are right, I can revert that

File: ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java (hunk 1 above)
Comment (justinlin-linkedin) [reply ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java:146]: the in progress compaction file is not included in the filter, so it won't even show up in the list.

**Review by JingQianCloud — APPROVED** [review]
approved

**Review by SophieGuo410 — APPROVED** [review]

**Pull request merged**